src/spu/spu.cc

```diff
@@ -302,16 +302,50 @@ inline int PCSX::SPU::impl::iGetNoiseVal(SPUCHAN *pChannel) {
 
     // mmm... depending on the noise freq we allow bigger/smaller changes to the previous val
     fa = pChannel->data.get<PCSX::SPU::Chan::OldNoise>().value +
-         ((fa - pChannel->data.get<PCSX::SPU::Chan::OldNoise>().value) / ((0x001f - ((spuCtrl & 0x3f00) >> 9)) + 1));
-    if (fa > 32767L) fa = 32767L;
+         ((fa - pChannel->data.get<PCSX::SPU::Chan::OldNoise>().value) / ((0x001f - ((spuCtrl & 0x3f00) >> 9)) + 1));*/
+
+    fa = (int16_t)dwNoiseVal;
+
+    /*if (fa > 32767L)
+        fa = 32767L;
     if (fa < -32767L) fa = -32767L;
-    pChannel->data.get<PCSX::SPU::Chan::OldNoise>().value = fa;
+    pChannel->data.get<PCSX::SPU::Chan::OldNoise>().value = fa;*/
 
     if (settings.get<Interpolation>() < 2)  // no gauss/cubic interpolation?
         SB[29].value = fa;                  // -> store noise val in "current sample" slot
     return fa;
 }
 
+// Noise Waveform - Dr. Hell (Xebra)
```

Comment:
If you've taken code from other people, please mention them in the AUTHORS file at the root.